lib/web_ui/lib/src/engine/html/picture.dart

```diff
@@ -581,9 +580,10 @@ class PersistedPicture extends PersistedLeafSurface {
   @override
   void retain() {
     super.retain();
-    final bool cullRectChangeRequiresRepaint = _computeOptimalCullRect(this);
-    if (cullRectChangeRequiresRepaint) {
+    _computeOptimalCullRect(this);
+    if (_requiresRepaint) {
       _applyPaint(this);
+      _requiresRepaint = false;
```

Comment:
This could be done in `_applyPaint` so it's cleared in all cases.

Reply:
Done.